plan/plan.go

```diff
@@ -179,43 +179,30 @@ func shouldUpdateTTL(desired, current *endpoint.Endpoint) bool {
 }
 
 func shouldUpdateProviderSpecific(desired, current *endpoint.Endpoint) bool {
-	if current.ProviderSpecific == nil && len(desired.ProviderSpecific) == 0 {
-		return false
-	}
-	for _, c := range current.ProviderSpecific {
-		// don't consider target health when detecting changes
-		// see: https://github.com/kubernetes-sigs/external-dns/issues/869#issuecomment-458576954
-		if c.Name == "aws/evaluate-target-health" {
-			continue
-		}
+	desiredProperties := map[string]endpoint.ProviderSpecificProperty{}
```

Comment:
I had pondered making this change, but decided not to because the number of properties is typically very small, so rather than a short iteration over an existing array, you've now got a memory allocation and some hashing happening. I'm not sure this is a win *in practice* — although I quite understand the urge to make this change, if the typical number of properties is only one or two at most, this is going to be slower.

Reply:
Might be but this code is now (I think( shorter and more readable, plus as you say the speed should not really matter as number of fields is generally low

Reply:
I could try to switch it back if maintainers really really want to

Reply:
Well I *didn't* say the speed didn't matter. Conceivably there are people out there with thousands of ingresses, for whom this might actually make a measurable difference. I don't *mind* the change, but I didn't make it because I realised it was probably better the way it was (even though I quite agree the code after this change is easier to read).